receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -53,8 +53,7 @@ func newReceiver(params receiver.Settings, config *Config, consumer consumer.Log
 
 	objects := make([]*K8sObjectsConfig, len(config.Objects))
 	for i, obj := range config.Objects {
-		copied := *obj // Copy the object
-		objects[i] = &copied
+		objects[i] = obj.DeepCopy()
```

Comment:
note: Since we create the `targetObjects` here then we don't deal with the `Config` object any more later in the receiver's execution. No extra "copies" are required later.